modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java

```diff
@@ -2316,31 +2323,50 @@ private void createNewStruct(int[] operands, StackFrame sf) {
         sf.refRegs[i] = bStruct;
     }
 
-    private void endTransaction(int[] operands) {
-        int i;
-        i = operands[0];
+    private void endTransaction(int status) {
         BallerinaTransactionManager ballerinaTransactionManager = context.getBallerinaTransactionManager();
         if (ballerinaTransactionManager != null) {
-            if (i == 0) {
+            if (status == 0) { //Transaction success
```

Comment:
Noted for change in future. We need a common place for all the VM level constants.